editor/modes/visual-editor/block.js

```diff
@@ -250,14 +250,18 @@ class VisualEditorBlock extends Component {
 	}
 
 	onFocus( event ) {
-		if ( event.target === this.node ) {
+		// Firefox retargets to parent with tabIndex.
+		const target = event.nativeEvent.explicitOriginalTarget || event.target;
```

Comment:
Safari also retargets, and no way to get the real target... :/ Maybe we're not supposed to put focusable elements inside a `target="0"`?

Reply:
Maybe @afercia knows more about this? :)

Reply:
I'm not sure nesting focusable elements is a good thing. It's like putting a link inside a link.

Reply:
Yeah, that's what I think too... This is no good: `<div tabindex="0"><button></button></div>`.

Reply:
Thanks @afercia for the quick feedback! I've removed the nesting in this branch now.